library/src/main/java/com/datatorrent/lib/io/jms/JMSTransactionableStore.java

```diff
@@ -65,17 +70,18 @@ public long getCommittedWindowId(String appId, int operatorId)
     logger.debug("Getting committed windowId appId {} operatorId {}", appId, operatorId);
 
     try {
-
       beginTransaction();
       BytesMessage message = (BytesMessage)consumer.receive();
-      logger.debug("Retrieved committed window message id {}", message.getJMSMessageID());
+      logger.debug("Retrieved committed window messageId: {}, messageAppOperatorIdProp: {}", message.getJMSMessageID(),
+          message.getStringProperty(APP_OPERATOR_ID));
       long windowId = message.readLong();
 
       message = getBase().getSession().createBytesMessage();
```

Comment:
Can you refactor these 4 lines to make a new function as follows. Then you can call it from 3 places (getCommittedWindow, storeCommitedWindow and connect)

private void writeWindowId(long windowId)
{
    BytesMessage message = getBase().getSession().createBytesMessage();
    message.setStringProperty(APP_OPERATOR_ID, appId + "_" + operatorId);
    message.writeLong(windowId);
    producer.send(message);
}